isis/appdata/export/TGOCaSSIS.tpl

```diff
@@ -0,0 +1,116 @@
+<?xml version="1.0" encoding="utf-8"?>
+<?xml-model href="http://pds.nasa.gov/pds4/disp/v1/PDS4_DISP_1700.sch" schemetypens="http://purl.oclc.org/dsdl/schematron"?>
+<?xml-model href="http://pds.nasa.gov/pds4/pds/v1/PDS4_PDS_1800.sch" schemetypens="http://purl.oclc.org/dsdl/schematron"?>
+<Product_Observational>
+ <CaSSIS_Header>
+  <IDENTIFICATION_DATA/>
+  <GEOMETRIC_DATA>
+   <PREDICTED_MAXIMUM_EXPOSURE_TIME unit="{{MainLabel.IsisCube.Archive.PredictMaximumExposureTime.Units}}">{{MainLabel.IsisCube.Archive.PredictMaximumExposureTime.Value}}</PREDICTED_MAXIMUM_EXPOSURE_TIME>
+   <CASSIS_OFF_NADIR_ANGLE unit="{{MainLabel.IsisCube.Archive.CassisOffNadirAngle.Units}}">{{MainLabel.IsisCube.Archive.CassisOffNadirAngle.Value}}</CASSIS_OFF_NADIR_ANGLE>
+   <PREDICTED_REQUIRED_REPETITION_FREQUENCY unit="{{MainLabel.IsisCube.Archive.PredictedRepetitionFrequency.Units}}">{{MainLabel.IsisCube.Archive.PredictedRepetitionFrequency.Value}}</PREDICTED_REQUIRED_REPETITION_FREQUENCY>
+   <TGO_GROUND_TRACK_VELOCITY unit="{{MainLabel.IsisCube.Archive.GroundTrackVelocity.Units}}">{{MainLabel.IsisCube.Archive.GroundTrackVelocity.Value}}</TGO_GROUND_TRACK_VELOCITY>
+   <FORWARD_ROTATION_ANGLE_REQUIRED unit="{{MainLabel.IsisCube.Archive.ForwardRotationAngle.Units}}">{{MainLabel.IsisCube.Archive.ForwardRotationAngle.Value}}</FORWARD_ROTATION_ANGLE_REQUIRED>
+   <SPICE_KERNEL_MISALIGNMENT_PREDICT unit="{{MainLabel.IsisCube.Archive.SpiceMisalignment.Units}}">{{MainLabel.IsisCube.Archive.SpiceMisalignment.Value}}</SPICE_KERNEL_MISALIGNMENT_PREDICT>
+  </GEOMETRIC_DATA>
+  <CaSSIS_General>
+   <TELESCOPE_FOCAL_LENGTH unit="{{MainLabel.IsisCube.Archive.FocalLength.Units}}">{{MainLabel.IsisCube.Archive.FocalLength.Value}}</TELESCOPE_FOCAL_LENGTH>
+   <TELESCOPE_F_NUMBER>{{MainLabel.IsisCube.Archive.FNumber.Value}}</TELESCOPE_F_NUMBER>
+  </CaSSIS_General>
+  <IMAGE_COMMAND/>
+  <FSW_HEADER>
+  <PEHK_HEADER>
+  <DERIVED_HEADER_DATA>
+   <PixelsPossiblySaturated>{{MainLabel.IsisCube.Archive.PixelsPossiblySaturated.Value}}</PixelsPossiblySaturated>
+   <Filter>{{MainLabel.IsisCube.BandBin.FilterName.Value}}</Filter>
+  </DERIVED_HEADER_DATA>
+ </CaSSIS_Header>
+ <Identification_Area>
+  <logical_identifier>urn:nasa:pds:TBD:TBD:TBD</logical_identifier>
+  <version_id>1.0</version_id>
+  <title>PDS4 product exported from ISIS3 cube.</title>
+  <information_model_version>1.8.0.0</information_model_version>
+  <product_class>Product_Observational</product_class>
+  <Modification_History>
+   <Modification_Detail>
+  </Modification_History>
+ </Identification_Area>
+ <Observation_Area>
+  <Time_Coordinates>
+   <start_date_time>{{MainLabel.IsisCube.Instrument.StartTime.Value}}</start_date_time>
+   {% if exists("MainLabel.IsisCube.Instrument.StopTime") %}
+   <stop_date_time>{{MainLabel.IsisCube.Instrument.StopTime.Value}}Z</stop_date_time>
+   {% else %}
+   <stop_date_time>{{MainLabel.IsisCube.Instrument.StartTime.Value}}Z</stop_date_time>
+   {% endif %}
```

Comment:
I tried setting up my if/else statements in my template in the same way and it caused blank lines in between the variables. Not sure if this was the case for you but it may be worth noting in case it causes issues with reading the label.

Reply:
Let's check with Tim. He placed a couple of ninja environment lines in his most recent PR for topds4. When I looked them up they sounded like they may be removing new lines and something else. Checkout  #4479 topds4 line 175, 176

Reply:
Also, blank lines shouldn't cause problems with the parsers, but a rendered template shouldn't have new-line characters in it that were not already there or did not come from the replacement string (i.e., json). Can we verify the new-lines are not in the json?

Reply:
So I confirmed that these new lines are being handled when processed to an isis label but I am not sure how they are handled when converting to an xml label.

Reply:
From the inja docs 

> While enabling set_trim_blocks removes the first newline after a statement, set_lstrip_blocks strips tabs and spaces from the beginning of a line to the start of a block.

This could also be handled by including a `-` in the syntax e.g. `{% else -%}` will clear whitespace around the tag according to the inja docs. Could be a better way to deal with this stuff rather than getting rid of leading/trailing whitespace wholesale.